### PR TITLE
Don't allow entity category "CONFIG" for sensors

### PR DESCRIPTION
**Pull request by edenhaus**
# What does this implement/fix?

The configuration category "config" is for read/write entities that control things. A sensor can never control things (and thus the change triggering because of this change are incorrect to begin with).
See https://github.com/home-assistant/core/pull/101471

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/sensor/__init__.py [hunk 1]
```diff
@@ -188,6 +189,17 @@ def validate_datapoint(value):
     return validate_datapoint({CONF_FROM: cv.float_(a), CONF_TO: cv.float_(b)})
 
 
+_SENSOR_ENTITY_CATEGORIES = {
+    k: v
+    for k, v in cv._ENTITY_CATEGORIES.items()  # pylint: disable=protected-access
```
Comment (jesserockz) [inline esphome/components/sensor/__init__.py]: Happy with the overall change, but can we just change this to be non-protected instead?
```suggestion
    for k, v in cv.ENTITY_CATEGORIES.items()
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/sensor/__init__.py (hunk 1 above)
Comment (edenhaus) [reply esphome/components/sensor/__init__.py]: Yes sure :) I thought that it was protected so everybody uses the function `entity_category` instead

File: esphome/components/sensor/__init__.py [hunk 2]
```diff
@@ -188,6 +189,17 @@ def validate_datapoint(value):
     return validate_datapoint({CONF_FROM: cv.float_(a), CONF_TO: cv.float_(b)})
 
 
+_SENSOR_ENTITY_CATEGORIES = {
+    k: v
+    for k, v in cv.ENTITY_CATEGORIES.items()  # pylint: disable=protected-access
```
Comment (jesserockz) [inline esphome/components/sensor/__init__.py]: Still needed?
```suggestion
    for k, v in cv.ENTITY_CATEGORIES.items()
```

**Review by jesserockz — DISMISSED** [review]
LGTM, Thanks

**Review by jesserockz — APPROVED** [review]

**Pull request merged**